crates/workers/src/router.rs

```diff
@@ -163,44 +209,77 @@ impl Chat for RouterModel {
                 let Some(model) = self.models.read().await.get(&name).map(Arc::clone) else {
                     return Err(OpenAIError::InvalidArgument(format!(
                         "Model router '{}' expects a model '{name}' to exist, but does not",
-                        self.router_name
+                        self.router_name.clone()
                     )));
                 };
 
-                model.chat_request(req).await
+                model.chat_request(req).instrument(span.clone()).await
             }
             Some(worker::RouterConfig::Weighted { from, .. }) => {
                 // `select_from_weighted` should be `Some(T)` since `self.models_cfg` isn't empty.
                 let name = select_from_weighted(&self.models_cfg).unwrap_or(from.clone());
+                tracing::info!(
+                    target: "task_history",
+                    progress = Progress::log().title(format!(
+                        "Worker '{}' deferring request to model '{name}'", self.router_name.clone()
+                        
+                    )).to_jsonl(),
+                );
                 let Some(model) = self.models.read().await.get(&name).map(Arc::clone) else {
                     return Err(OpenAIError::InvalidArgument(format!(
                         "Model router '{}' expects a model '{name}' to exist, but does not",
-                        self.router_name
+                        self.router_name.clone()
                     )));
                 };
 
-                model.chat_request(req).await
+                model.chat_request(req).instrument(span.clone()).await
             }
             Some(worker::RouterConfig::Fallback { .. }) => {
                 let fallbacks = into_ordered_fallbacks(&self.models_cfg);
                 for (name, _) in fallbacks {
+                    tracing::info!(
+                        target: "task_history",
+                        progress = Progress::log().title(format!(
+                            "Worker '{}' deferring request to model '{name}'", self.router_name.clone()
+                        )).to_jsonl(),
+                    );
                     let Some(model) = self.models.read().await.get(&name).map(Arc::clone) else {
                         return Err(OpenAIError::InvalidArgument(format!(
                             "Model router '{}' expects a model '{name}' to exist, but does not",
-                            self.router_name
+                            self.router_name.clone()
                         )));
                     };
 
-                    if let Ok(resp) = model.chat_request(req.clone()).await {
-                        return Ok(resp);
+                    match model.chat_request(req.clone()).instrument(span.clone()).await {
+                        Err(e) => {
+                            tracing::error!(
+                                target: "task_history",
+                                progress = Progress::log()
```

Comment:
```suggestion
                                progress = Progress::error()
```

Reply:
It feels a little weird that the progress verbosity does not match the tracing verbosity

Reply:
Maybe `log` is not the best name. `Progress::error`, etc are meant to be type of progress messages moreso than verbosity